site-docker.yml.sample

```diff
@@ -32,7 +32,7 @@
       delegate_facts: True
       with_items: "{{ groups['all'] }}"
       when:
-        - delegate_facts_host | bool
+        - delegate_facts_host | bool or inventory_hostname not in groups.get('clients')
```

Comment:
2 things:

1. The logic seems wrong:
This will delegate facts if delegate_facts_host is true or the host isnt in the clients group. E,g, if you set "delegate_facts_host = false" it will still delegate on non-client hosts.
The title makes it sound like you want delegate_facts_host and inventory_hostname not in groups.get('clients') - which should delegate if delegate_facts_host is true and the host isn't in the client group.

2. This has a consequence because if you have a host that is in the clients group and another group (for whatever reason) it won't delegate facts. You may want to check if it is in all the other groups, but not in the clients group, but that makes a long statement.

Maybe a better way would be to remove the "clients" from the hosts: for this task (line 13), then the gather facts task wont run against clients though, you could run that first against clients.

Reply:
that's correct, i'll update this commit.
thanks @andymcc

Reply:
I like the idea of removing ``clients`` from the ``hosts`` line for this play. Perhaps we should make this fact delegation it's own play entirely, not add ``clients`` there but then keep them here so that ``ceph-defaults`` and ``ceph-docker-common`` can be run against them?

I think that would take care of the second concern @andymcc had.